src/application/Export.java

```diff
@@ -1,39 +1,148 @@
 package application;
 
+import data.Group;
+import data.Item;
 import data.Lesson;
 
 import javax.swing.*;
 import javax.xml.bind.JAXBContext;
 import javax.xml.bind.JAXBException;
 import javax.xml.bind.Marshaller;
 import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
 
 public class Export {
 
     private String targetPath;
+    private ArrayList<Lesson> xml;
     private FileManager fm;
 
     public Export() {
         fm = new FileManager();
     }
 
-    private void chooseTargetPath() {
-        JButton open = new JButton();
-        JFileChooser fo = new JFileChooser();
-        fo.setCurrentDirectory(new java.io.File("C:"));
-        fo.setDialogTitle("hello");
-        fo.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
-        if(fo.showOpenDialog(open) == JFileChooser.APPROVE_OPTION){
-            //
+    private void chooseTargetPath() throws IOException {
+        try {
+            JButton open = new JButton();
+            JFileChooser fo = new JFileChooser();
+            fo.setCurrentDirectory(new java.io.File("C:"));
+            fo.setDialogTitle("chooser");
+            fo.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+            if (fo.showOpenDialog(open) == JFileChooser.APPROVE_OPTION) {
+                //
+            }
+            if (fo == null) {
```

Comment:
nepadlo by to uz davnejsie na volaniach metod objektu fo (ako napr 'fo.setDialogTitle()') ak by to bolo null? nieje tento if nepotrebny?